pkg/metricprovider/external_metric_provider.go

```diff
@@ -57,65 +56,69 @@ const (
 func (p *ExternalMetricProvider) GetExternalMetric(ctx context.Context, namespace string, metricSelector labels.Selector, info provider.ExternalMetricInfo) (*external_metrics.ExternalMetricValueList, error) {
 	klog.Info(fmt.Sprintf("Get metric by selector for external metric, Info %v namespace %s metricSelector %s", info, namespace, metricSelector.String()))
 
-	if !IsLocalExternalMetric(info, p.client) {
-		if p.remoteAdapter != nil {
-			return p.remoteAdapter.GetExternalMetric(ctx, namespace, metricSelector, info)
-		} else {
-			return nil, apiErrors.NewServiceUnavailable("not supported")
-		}
-	}
-
-	if strings.HasPrefix(info.Metric, "crane_cron") {
+	switch info.Metric {
+	case known.MetricNameCron:
 		return p.GetCronExternalMetrics(ctx, namespace, metricSelector, info)
-	}
-
-	if strings.HasPrefix(info.Metric, "crane") {
-		prediction, err := GetPrediction(ctx, p.client, namespace, metricSelector)
+	case known.MetricNamePrediction:
+		predictions, err := GetPredictions(ctx, p.client, namespace, metricSelector)
 		if err != nil {
 			return nil, err
 		}
 
-		timeSeries, err := utils.GetReadyPredictionMetric(info.Metric, prediction)
-		if err != nil {
-			return nil, err
-		}
-
-		// get the largest value from timeSeries
-		// use the largest value will bring up the scaling up and defer the scaling down
-		timestampStart := time.Now()
-		timestampEnd := timestampStart.Add(time.Duration(prediction.Spec.PredictionWindowSeconds) * time.Second)
-		largestMetricValue := &metricValue{}
-		hasValidSample := false
-		for _, v := range timeSeries.Samples {
-			// exclude values that not in time range
-			if v.Timestamp < timestampStart.Unix() || v.Timestamp > timestampEnd.Unix() {
-				continue
+		for _, prediction := range predictions {
+			resourceIdentifier, bl := metricSelector.RequiresExactMatch("resourceIdentifier")
```

Comment:
what's meaning for `bl` ? better to change to `found` for all metricSelector.RequiresExactMatch(xxxx)

Reply:
change utils.GetReadyPredictionMetric(info.Metric, prediction) to 
utils.GetReadyPredictionMetric(info.Metric, resourceIdentifier, &prediction)
,need to get resourceIdentifier, and same as resourceIdentifier == ""